spec/CookieSpec.php

```diff
@@ -239,6 +239,38 @@ function it_matches_other_cookies()
         $this->match($notMatches)->shouldReturn(false);
     }
 
+    function it_validates_itself()
+    {
+        $this->isValid()->shouldReturn(true);
+    }
+
+    /**
+     * @dataProvider invalidCharacterExamples
```

Comment:
Not sure if we want to test the fact that validation actually works correctly here. Although I can't really see any tests for that. Nevertheless I think we only need to test two cases here: validation passes/validation fails. For that I am not sure if we need data provider. In fact using data providers in specification tests is kind of abusing BDD, these kind of tests should be done in separate unit tests.